presto-main/src/main/java/com/facebook/presto/operator/aggregation/AggregationCompiler.java

```diff
@@ -119,6 +124,13 @@ public static BindableAggregationFunction generateAggregationBindableFunction(Cl
         return builder.build();
     }
 
+    private static List<LongVariableConstraint> parseLongVariableConstraints(Method inputFunction)
```

Comment:
This needs tests. Can you either port an existing aggregation, or add one for testing?